### DISCUSSION TITLE
Qodana Teamcity Integration seems not to work properly when agents are running in docker

### DISCUSSION THREAD

**Discussion by Sp0tlight** [JetBrains/Qodana#122] [Q&A]
Our TeamCity and all agents are running via individual docker containers on a common host.

The default Qodana Teamcity integration (Build Step -> Qodana) therefore requires it that the working directory is synced from the agent to the host, so that the Qodana docker image that is spinned up via TC can access it:

```
Agent (docker volume mount):
<HOST>/tc/agent-1/work:/opt/buildagent/work

Qodana build step config:
Working directory=<HOST>/tc/agent-1/work/checkoutDirID/
```

Currently we encounter 2 issues:

1) We can't set the result directory as it is already mapped internally by the command and therefore the internal default one pollutes the host. This docker volume mapping is used internally by the integration plugin: 
```
/opt/buildagent/temp/buildTmp/Static_Analysis_Results/Code_Inspection:/data/results/
``` 
so the results will be written directly on the host at `<HOST>/opt/buildagent/temp/buildTmp/Static_Analysis_Results/Code_Inspection`. Here we would expect some additional config option like the working directory, so we can redirect the output to the host mapped agent volume.

2) Qodana don't report anything at all in TeamCity but rather fails after successful analysis, as it can't resolve the written results.

### Analysis is running successfully
> [22:13:09 ]()    Qodana - Detailed summary
> [22:13:09 ]()    Analysis results: 238 problems detected
> [22:13:09 ]()    By severity: Critical - 11, High - 217, Moderate - 10
> [22:13:09 ]()    ----------------------------------------------------------------------------
> [22:13:09 ]()    Name                                                Severity  Problems count
> [22:13:09 ]()    ----------------------------------------------------------------------------
> [22:13:09 ]()    Usage of API marked for removal                     Critical              11
> [22:13:09 ]()    Constant conditions & exceptions                        High              17
> [.........]

### But completion and TeamCity report is failing
> [22:13:12 ]()    Generating final reports...
> [22:13:13 ]()    The project doesn't contain qodana.yaml config file
> [22:13:13 ]()    Done
> [22:13:13 ]()    2022/08/24 20:13:13 Sync IDEA cache from: /data/project/.idea to: /data/cache/.idea
> [22:13:14 ]()    Docker exit code: 0
> [22:13:14 ]()  **Can't resolve .descriptions.json** << error
> [22:13:14 ]()  Process exited with code 0
> [22:13:14 ]()  Reporting as tests is enabled
> [22:13:14 ]()  Mode: test on each problem
> [22:13:14 ]()  Reported as tests: 0 problems
> [22:13:14 ]()  **No one instrument is enabled, or all enabled have failed** << error

The file itself exists on the host, so it's probably some issue in the way that the qodana image wants to read the results for reporting it to TeamCity.

 `ls -a <HOST>/opt/buildagent/temp/buildTmp/Static_Analysis_Results/Code_Inspection | grep .des`
```
.descriptions.json
```

**Comment by tiulpin** [comment]
Hi! I created an issue in our tracker – https://youtrack.jetbrains.com/issue/QD-3932/

Thank you for creating a GitHub issue from discussions, but we keep all our issues in our tracker, though we still use discussions.

So it's OK to continue solving the issue here or in YouTrack as you wish, we just don't use GitHub issues

**Comment by tiulpin** [comment]
@Sp0tlight could you please check that you configured the agents properly, as specified here - https://hub.docker.com/r/jetbrains/teamcity-agent - in  Running Builds Which Require Docker section?

Meanwhile we will try investigate it more from our side. Thanks for reporting!

**Reply by Sp0tlight** [reply]
@tiulpin Thank you for creating the issue. And yeah, everything is setup correctly.

**Reply by eddiebeazer** [reply]
Was there an update on this?  I'm also having the same issue with the JS and Go containers using a team city docker agent
